### DISCUSSION TITLE
What is the recommended way to reprompt a user in a node until entered data is valid?

### DISCUSSION THREAD

**Discussion by sehranhas** [microsoft/promptflow#3311] [Q&A]
I have a use case where I would like to reprompt the user to enter some details until all necessary information is passed. How do you make promptflow reprompt the user to enter all the necessary details until moving to the next node and how does the next node validate the entered data is valid? Right now promptflow automatically executes the next step as soon as a chat node or script node is done. I also see that promptflow does not support the Python `input()` call so I cannot pass it into a `while` loop until all fields are entered and valid.

**Comment by brynn-code** [comment]
Hi @sehranhas, thanks for reaching us. I'm sorry that promptflow does not support `while` loop for now, this does needs a `while` loop to achieve. Maybe you could try put the while loop with `input()` inside one python script node in flow, but that's also a limited scenario, we assume it only works if you run flow in command without UI.
